### PR TITLE
Changed default orientation to Auto for TS100, TS80(P) & Pinecil ...

### PR DESCRIPTION
**Pull request by discip**
... as requested here https://github.com/Ralim/IronOS/issues/1157.

Also some minor changes to **`gui.cpp`**.

### CODE REVIEW

**Comment by Ralim** [comment]
Hi,
Would rather just leave the inverted screen there for now since we have the room, but it's one of the first on the chopping block when we need room ?

**Comment by discip** [comment]
Good morning Sir,
as mentioned above and you most likely have observed it for yourself already, the code is still in there and I did not plan to remove it.

However I would like to keep it hidden like that, until it is needed.

Do you mean to revert the change, or was your concern just about removing the code entirely?


Are you good with the change of the default orientation?

**Comment by Johnex** [comment]
I like the inverted screen, can we keep it?

**Comment by discip** [comment]
@Johnex
Of course, this is why I asked if someone would miss is. 😊

**Comment by discip** [comment]
@Ralim
If everyone is happy, also regarding *default orientation to Auto*, this should be good to go. 😃

kind regards

File: source/Core/Inc/Settings.h [hunk 1]
```diff
@@ -47,8 +47,8 @@ enum SettingsOptions {
   PDMissingWarningCounter        = 30, // Counter of how many times we have warned we cannot detect the pd interface
   UILanguage                     = 31, // Selected UI Language code, null-terminated *only if* the length is less than 8 chars
   PDNegTimeout                   = 32, // PD timeout in 100ms steps
-  OLEDInversion                  = 33, // Invert the colours on the display
```
Comment (Ralim) [inline source/Core/Inc/Settings.h:50]: Can you undo this swap please, need to keep setting <-> number the same to support settings migration on updates :)

File: source/Core/Src/Settings.cpp [hunk 2]
```diff
@@ -82,8 +82,8 @@ static const SettingConstants settingsConstants[(int)SettingsOptions::SettingsOp
     {0, 9, 1, 0},                                                   // PDMissingWarningCounter
     {0, 0xFFFF, 0, 41431 /*EN*/},                                   // UILanguage
     {0, 50, 1, 20},                                                 // PDNegTimeout
-    {0, 1, 1, 0},                                                   // OLEDInversion
```
Comment (Ralim) [inline source/Core/Src/Settings.cpp:85]: Undo the swap here too please :)

File: source/Core/Threads/GUIThread.cpp [hunk 3]
```diff
@@ -553,9 +553,9 @@ static void gui_solderingMode(uint8_t jumpToSleep) {
 #else
     if (OLED::getRotation()) {
 #endif
-      OLED::setCursor(50, 0);
```
Comment (Ralim) [inline source/Core/Threads/GUIThread.cpp:556]: What was this for?

**Review by Ralim — CHANGES_REQUESTED** [review]

**Comment by Ralim** [comment]
If you can please swap those settings entries back would be great so we dont mess with settings on upgrade :)

What was the gui tweak for?
@discip

**Comment by discip** [comment]
@Ralim
Should be good now. 😃

Sorry for turning knobs not knowing what they are meant for. 😅

File: source/Core/Inc/Settings.h (hunk 1 above)
Comment (discip) [reply source/Core/Inc/Settings.h]: Done. 😊

File: source/Core/Src/Settings.cpp (hunk 2 above)
Comment (discip) [reply source/Core/Src/Settings.cpp]: Done. 😊

File: source/Core/Threads/GUIThread.cpp (hunk 3 above)
Comment (discip) [reply source/Core/Threads/GUIThread.cpp]: Sorry, that was me not realizing these changes would also be pulled. 😓

**Comment by Johnex** [comment]
Auto would be great, i usually set it to auto first thing i do 👍

**Comment by Ralim** [comment]
@discip No problem at allll, that is what I'm here to check 😁

**Comment by discip** [comment]
@Ralim
Good afternoon,
coming back to this:
> What was the gui tweak for?

https://github.com/Ralim/IronOS/blob/26493c0104578cdcfe7ed4b5389e7d18b7d4027a/source/Core/Threads/GUIThread.cpp#L558-L563
Why I am not able to shift this down by a pixel?

It results in showing only the lower half of the digits. 🤷‍♂️😵

thanks in advance

**Comment by Ralim** [comment]
To remove a bug chunk of complexity, the screen only supports vertical offsets of 0 or 8 pixels, as otherwise we have to shift every bit in every byte along one, and wrap overflow to the next row.

So it gets really messy 😅

By only supporting the two height offsets (0/8) it removes a big chunk of code that I didn't feel like writing

**Comment by discip** [comment]
Thank you for the explanation. 😃👍
Is there an elegant way of shifting characters horizontally nonetheless?

I would need to shift the large digits down by only 1 pixel to make them align properly.

**Comment by Ralim** [comment]
You can move things horizontally on the screen by one pixel increments no issue.
But height is locked to the 0/8 (top row, bottom row).

If it is _all_ digits, it is better to shift the fontmap

**Comment by discip** [comment]
Sorry, I meant **_vertically_** of course.

Where would I the fontmap?
And  what exactly would I have to change?

**Comment by Ralim** [comment]
Soo......

The font table is here: https://github.com/Ralim/IronOS/blob/master/Translations/font_tables.py

In each byte, the LSB (bit0) is the top of that column, and the MSB (bit1) is at the bottom of that column.

So to move a symbol "down" a pixel, you want to: 
1. For each byte in the second half of the set of bytes (ie. bottom row) shift all the bits left by one (up) <<1
2. For each byte in the first half of the set of bytes (i.e. top row) copy their MSB (top bit) into the LSB (bottom bit) of each matching byte in the second half of the set of bytes
3. For each byte in the first half of the set of bytes, shift all the bits left by one (up) <<1

If that is too messy to understand I can try and write some quite python to do the shifts for you.
But probably best to split this out to an issue with explanation of what the goal is :)

**Comment by discip** [comment]
Thank you for the thorough explanation.
But as you might have guessed, this is to high for me, to wrap my head around, especially in code.

Since I thought of changing all the characters by hand, your proposal is highly welcomed.

So you planned on writing a script, that is run once and that would change all the actual characters in the font map. Right?

And yes I will open a new issue on that.
But I have to pause this for now and plan on coming back on Monday.

Have a nice weekend.

**Comment by discip** [comment]
@Ralim
Good morning Sir,
> If it is *all* digits, it is better to shift the fontmap

I just noticed that changing the font table, would harm other scenarios, so I unfortunately have to either drop the entire idea, or you help me to make it work for only some cases. 😁

**Pull request merged**